.github/scripts/test_record_expiry.py

```
@@ -0,0 +1,317 @@
+        self.assertIsNotNone(self.recorded(MAC))
+
+
+class MalformedStateTest(TermTestCase):
+    """A corrupt term must stop the run, never propagate to the roster."""
+
+    def test_an_unparseable_recorded_term_is_refused(self):
+        """Copied verbatim it would read as 'no expiry' — an unlimited licence."""
+        self.bind(MAC, term="not-a-date")
+        self.bind(WIN)
+
+        with self.assertRaises(SystemExit):
```

Ruff (0.16.3)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)